### PR TITLE
Notifier implementation

### LINKED ISSUE #82: Notifier
This class sends emails when a job is completed.

### PR DESCRIPTION
**Pull request by ammen99**
This is a basic implementation of the email notifier. Contains commits from other PRs, the only relevant commits are the last 2 (or maybe only 1?).

I have added unit tests to check that emails are attempted to be sent that contain the correct message and recipient. During the integration phase we'll likely want to add a dummy server to test the real sending of emails (I tested it locally with gmail, seems to work) 

Fixes #82

### CODE REVIEW

File: src/ja/server/email.py [hunk 1]
```diff
@@ -1,23 +1,99 @@
...
+        @param subject The subject of the email.
+        @param contents The contents of the email.
+        @param send_to The email address to send the message to.
+        """
+
+
+class BasicEmailServer(EmailServerBase):
+    _EMAIL_SENDER = "no-reply@jobadder.com"
+
+    def __init__(self, smpt_server_address: str, smtp_server_port: int, smtp_user: str, smtp_password: str) -> None:
```
Comment (nikolatzotchev) [inline src/ja/server/email.py]: simple mail protocol transfer
smpt -> smtp :)

File: src/ja/server/email.py [hunk 2]
```diff
@@ -1,23 +1,99 @@
...
-    EmailNotifier is responsible for listening for changes in the database and sending emails to the owners of jobs
-    when their jobs finish running.
+    EmailNotifier is responsible for sending emails to the owners of a job when the job is finished running.
     """
-    def __init__(self, database: ServerDatabase):
+    def __init__(self, email_server: EmailServerBase):
         """!
         Initialize the email notifier, and set the callback for the given database.
 
         @param database The database to notify jobs for.
```
Comment (nikolatzotchev) [inline src/ja/server/email.py]: maybe fix this no database -> email_server

File: src/ja/server/email.py (hunk 1 above)
Comment (nikolatzotchev) [reply src/ja/server/email.py]: fix

File: src/test/server/test_email.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        self._unit.assertIsNotNone(send_to)
+
+        self._unit.assertEqual(subject, self._subject)
+        self._unit.assertEqual(contents, self._msg)
+        self._unit.assertEqual(send_to, self._email)
+        self._subject = self._msg = self._email = None
+
+
+class EmailNotifierTest(TestCase):
+    EMAIL = 'zemisemki@hotmail.com'
```
Comment (nikolatzotchev) [inline src/test/server/test_email.py]: nice email

File: src/ja/server/email.py [hunk 4]
```diff
@@ -1,23 +1,99 @@
...
+            print(e)
+
+    def __del__(self) -> None:
+        if self._email_client:
+            self._email_client.close()
+
+    def send_email(self, subject: str, contents: str, send_to: str) -> None:
+        if not self._email_client:
+            return
+        print("start send email")
```
Comment (nikolatzotchev) [inline src/ja/server/email.py]: why do you need this print

Comment (ammen99) [reply src/ja/server/email.py]: leftover from debugging, removed now.

File: src/ja/server/email.py (hunk 2 above)
Comment (ammen99) [reply src/ja/server/email.py]: fixed

**Comment by ammen99** [comment]
Addressed review comments by @nikolatzotchev

**Pull request merged**